snowpack/src/build/process.ts

```diff
@@ -0,0 +1,365 @@
+  const isDev = !!isWatch;
+  const isSSR = !!config.buildOptions.ssr;
+  const isHMR = getIsHmrEnabled(config);
+
+  // Seems like maybe we shouldn't be doing this...
+  config.buildOptions.resolveProxyImports = !config.optimize?.bundle;
+  config.devOptions.hmrPort = isHMR ? config.devOptions.hmrPort : undefined;
+  config.devOptions.port = 0;
+
+  const clean = config.buildOptions.clean;
+  const buildDirectoryLoc = config.buildOptions.out;
+
+  const devServer = await startServer(commandOptions, {isDev, isWatch, preparePackages: false});
+
+  return {
+    commandOptions,
+    config,
+    isDev,
+    isHMR,
+    isSSR,
+    isWatch,
+    clean,
+
+    buildDirectoryLoc,
+    allBareModuleSpecifiers: [],
+    allFileUrlsUnique: new Set<string>(),
+    allFileUrlsToProcess: [],
+    devServer
+  };
+}
+
+export function maybeCleanBuildDirectory(state: BuildState) {
+  const {buildDirectoryLoc} = state;
+  if (state.clean) {
+    deleteFromBuildSafe(buildDirectoryLoc, state.config);
+  }
+  mkdirp.sync(buildDirectoryLoc);
+}
+
+export async function addBuildFiles(state: BuildState, files: string[]) {
+  const {config} = state;
+
+  const excludePrivate = new RegExp(`\\${path.sep}\\..+(?!\\${path.sep})`);
+  const excludeGlobs = [...config.exclude, ...config.testOptions.files];
+  const foundExcludeMatch = picomatch(excludeGlobs);
+
+  const allFileUrls: string[] = [];
+
+  for (const f of files) {
+    if (excludePrivate.test(f) || foundExcludeMatch(f)) {
+      continue;
+    }
+    const fileUrls = getUrlsForFile(f, config)!;
+    // Only push the first URL. In multi-file builds, this is always the JS that the
+    // CSS is imported from (if it exists). That CSS may not exist, and we don't know
+    // until the JS has been built/loaded.
+    allFileUrls.push(fileUrls[0]);
+  }
+
+  state.allBareModuleSpecifiers = [];
+  state.allFileUrlsUnique = new Set(allFileUrls);
+  state.allFileUrlsToProcess = [...state.allFileUrlsUnique];
+}
+
+export async function addBuildFilesFromMountpoints(state: BuildState): Promise<void> {
+  const {config} = state;
+
+  const possibleFiles: string[] = [];
+
+  for (const [mountKey, mountEntry] of Object.entries(config.mount)) {
+    logger.debug(`Mounting directory: '${mountKey}' as URL '${mountEntry.url}'`);
+    const files = (await new fdir().withFullPaths().crawl(mountKey).withPromise()) as string[];
+
+    possibleFiles.push(...files);
+  }
+  
+  addBuildFiles(state, possibleFiles);
+}
+
+type FlushLoadOptions = LoadUrlOptions & {encoding?: undefined};
+
+async function flushFileQueue(state: BuildState, ignorePkg: boolean, loadOptions: FlushLoadOptions) {
+  const {
+    config,
+    allFileUrlsUnique,
+    allFileUrlsToProcess,
+    allBareModuleSpecifiers,
+    buildDirectoryLoc,
+    devServer,
+    isHMR
+  } = state;
+
+  const pkgUrlPrefix = path.posix.join(config.buildOptions.metaUrlPath, 'pkg/');
+
+  logger.debug(`QUEUE: ${allFileUrlsToProcess}`);
+  while (allFileUrlsToProcess.length > 0) {
```

Comment:
Just out of curiosity: I know this was 1:1 in the original version, but should we be doing these in parallel instead of one-by-one?

Reply:
Not sure, would prefer to do any changes in another PR though.